lib/text_count_util/counter.ex

```diff
@@ -0,0 +1,36 @@
+defmodule TextCountUtil.Counter do
+  use GenServer
+
+  alias TextCountUtil.{CounterDynamicSupervisor, CounterRegistry}
+
+  def start_link(line_number) do
+    name = {:via, Registry, {CounterRegistry, line_number}}
+    DynamicSupervisor.start_child(
+      CounterDynamicSupervisor,
+      %{
+        id: GenServer,
+        start: {
+          GenServer,
+          :start_link,
+          [__MODULE__, 0, [name: name]]
+        },
+        restart: :transient
+      }
+    )
+  end
+
+  ## Callbacks
+
+  @impl GenServer
+  def init(_), do: {:ok, 0}
```

Comment:
As per my comment bove.

```suggestion
  def init(line), do: {:ok, 0, {:continue, {:count, line}}}
  
  @impl true
  def handle_continue({:count, line}, 0), do: {:noreply, Enum.count(…)}
```